src/processors/models/processorsManager.js

```diff
@@ -9,6 +9,10 @@ const logger = require('../../common/logger'),
 
 module.exports.createProcessor = async function (processor) {
     let processorId = uuid.v4();
+    const processorWithTheSameName = await databaseConnector.getProcessorByName(processor.name);
```

Comment:
check/throw error first in the function before the `let processorId = uuid.v4();`

Reply:
done